src/Microsoft.PowerShell.Commands.Management/commands/management/Service.cs

```diff
@@ -1756,7 +1859,31 @@ protected override void ProcessRecord()
                                 ErrorCategory.PermissionDenied);
                         }
 
+                        // Set the delayed auto start
+                        NativeMethods.SERVICE_DELAYED_AUTO_START_INFO ds = new NativeMethods.SERVICE_DELAYED_AUTO_START_INFO();
+                        ds.fDelayedAutostart = StartupType == ServiceStartupType.AutomaticDelayedStart;
+                        size = Marshal.SizeOf(ds);
+                        buffer = Marshal.AllocCoTaskMem(size);
+                        Marshal.StructureToPtr(ds, buffer, false);
```

Comment:
I wonder where we destroy and free?

Reply:
@daxian-dbw Could you please comment?

Reply:
Good catch! I don't find the allocated memory is freed anywhere. `FreeCoTaskMem` should be called once done using the memory. The same issue also happens in some other places in this file.

Reply:
I found `AllocCoTaskMem` in 4 files and ` FreeCoTaskMem` in 17 files - It seems we should widely review. Maybe don't block the PR and open a tracking Issue for new PR?